backend/src/main/java/com/agh/introwertycznelosie/data/Room.java

```diff
@@ -0,0 +1,146 @@
+package com.agh.introwertycznelosie.data;
+
+import org.springframework.data.domain.Range;
+
+import javax.persistence.*;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@Entity
+public class Room {
+
+    @Id
+    @GeneratedValue
+    private int id;
+
+    private int recommendedCapacity;
+    private int maximalCapacity;
+    private String localization;
+    private int number;
+    @OneToMany(mappedBy = "room")
+    private List<DateRange> availableDates = new ArrayList<>();
+
+    public int getId() {
+        return id;
+    }
+
+    public void setId(int id) {
+        this.id = id;
+    }
+
+    public int getRecommendedCapacity() {
+        return recommendedCapacity;
+    }
+
+    public void setRecommendedCapacity(int recommendedCapacity) {
+        this.recommendedCapacity = recommendedCapacity;
+    }
+
+    public int getMaximalCapacity() {
+        return maximalCapacity;
+    }
+
+    public void setMaximalCapacity(int maximalCapacity) {
+        this.maximalCapacity = maximalCapacity;
+    }
+
+    public String getLocalization() {
+        return localization;
+    }
+
+    public void setLocalization(String localization) {
+        this.localization = localization;
+    }
+
+    public int getNumber() {
+        return number;
+    }
+
+    public void setNumber(int number) {
+        this.number = number;
+    }
+
+    public List<DateRange> getAvailableDates() {
+        return availableDates;
+    }
+
+    public void setAvailableDates(List<DateRange> availableDates) {
+        this.availableDates = availableDates;
+    }
+
+    @Entity
+    private static class DateRange {
```

Comment:
Nie wiem czy nie lepiej by było, gdyby to była zewnętrzna klasa - np w momencie gdy będziemy chcieli znaleźć wolną salę w danym terminie

Reply:
46db096dad8496b77aa10568f84456387549050f